homework8/task01.py

```diff
@@ -0,0 +1,118 @@
+"""
+We have a file that works as key-value storage, each line is represented
+as key and value separated by = symbol, example:
+
+name=kek last_name=top song_name=shadilay power=9001
+
+Values can be strings or integer numbers. If a value can be treated both
+as a number and a string, it is treated as number.
+
+Write a wrapper class for this key value storage that works like this:
+
+storage = KeyValueStorage('path_to_file.txt') that has its keys and values
+accessible as collection items and as attributes.
+Example:
+     storage['name'] # will be string 'kek' storage
+     song_name # will be 'shadilay' storage
+     power # will be integer 9001
+
+In case of attribute clash existing built-in attributes take precedence.
+In case when value cannot be assigned to an attribute
+(for example when there's a line 1=something) ValueError should be raised.
+File size is expected to be small, you are permitted to read it entirely into
+memory.
+"""
+from typing import Generator, Tuple, Union
+
+
+def line_gen(
+        path: str, encoding: str = 'utf-8', errors: str = 'ignore'
+) -> Generator:
+    """
+    Read file line by line
+
+    :param path: Path to the file
+    :type path: str
+    :param encoding: Codec for encoding (more info python build-in open)
+    :type encoding: str
+    :param errors: Way to handle errors (more info python build-in open)
+    :type errors: str
+    :return: Generator yielding lines from file
+    :rtype: Generator
+    """
+    with open(path, 'r', encoding=encoding, errors=errors) as fi:
+        line = fi.readline()
+        while line and '\n' in line:
+            yield line
+            line = fi.readline()
+
+
+def convert_to_int(input_value: str) -> Union[str, int]:
+    """
+    Try to convert input string value to integer
+
+    :param input_value: Input string
+    :type input_value: str
+    :return: Integer if conventing is possible else initial string
+    :rtype: Union[str, int]
+    """
+    try:
+        return int(input_value)
+    except ValueError:
+        return input_value
+
+
+def process_line(line: str) -> Tuple[Union[int, str], Union[int, str]]:
+    """
+    Split line to key and value pairs
+
+    :param line: String that looks like "key=value"
+    :type line: str
+    :return: Key and value from line
+    :rtype: Tuple[Union[int, str], Union[int, str]]
+    """
+    line = line.rstrip('\n')
+    key, value = line.split('=')
+    return convert_to_int(key), convert_to_int(value)
+
+
+class KeyValueStorage:
```

Comment:
file.txt
```__class__=10```

---------------------
```
storage = KeyValueStorage("file.txt")
storage["__class__"], storage.__class__   #  10, KeyValueStorage
```

as u can see, we get different results for one attr, it's confusiong a bit. so u have 2 choises:
1) dont let to get (built-in) magic attrs by ["<attr>"]
2) return same values